java/src/test/java/com/google/oak/client/OakClientTest.java

```diff
@@ -16,114 +16,76 @@
-    PilotOakClient oakClient = new PilotOakClient.Builder(attestationClient).build();
-    Result<String, Exception> response = oakClient.send("Hello!");
-    assertTrue(response.isSuccess());
-    assertEquals("Hello!", response.success().get());
-  }
-
-  private static class PilotOakClient implements OakClient<String, String> {
-    final PilotRpcClient rpcClient;
-    final Encryptor encryptor;
-
-    PilotOakClient(final Builder builder) {
-      byte[] signingPublicKey = builder.attestationClient.verifyEvidence(new Evidence() {});
-      // In reality implementations of OakClient must handle empty Optionals and unexpected types.
-      rpcClient = builder.attestationClient.getRpcClient().success().get();
-      encryptor = builder.attestationClient.getEncryptor(signingPublicKey).success().get();
-    }
-
-    @Override
-    public void close() throws Exception {
-      rpcClient.close();
-    }
-
-    @Override
-    public Result<String, Exception> send(final String request) {
-      return encryptor.encrypt(request.getBytes())
-          .andThen(rpcClient::send)
-          .andThen(encryptor::decrypt)
-          .map(b -> new String(b, StandardCharsets.UTF_8));
-    }
-
-    static class Builder {
-      final PilotAttestationClient attestationClient;
-
-      Builder(final PilotAttestationClient attestationClient) {
-        this.attestationClient = attestationClient;
-      }
-
-      PilotOakClient build() {
-        return new PilotOakClient(this);
-      }
-    }
-  }
-
-  private static class PilotAttestationClient
-      implements OakClient.EncryptorProvider<Encryptor>,
-                 OakClient.RpcClientProvider<PilotRpcClient> {
-    final PilotRpcClient rpcClient;
-    final Encryptor encryptor = new Encryptor() {
-      @Override
-      public Result<byte[], Exception> encrypt(final byte[] data) {
-        return Result.success(data);
-      }
-
-      @Override
-      public Result<byte[], Exception> decrypt(final byte[] data) {
-        return Result.success(data);
-      }
-    };
-
-    PilotAttestationClient() {
-      this.rpcClient = new PilotRpcClient();
-    }
-
-    // This function will eventually be provided by an EvidenceProvider.
-    byte[] verifyEvidence(Evidence evidence) {
-      return "TestSigningPublicKey".getBytes(StandardCharsets.UTF_8);
-    }
+  private static final byte[] TEST_SERVER_ENCRYPTION_PUBLIC_KEY = new byte[0];
+  private static final byte[] TEST_REQUEST = new byte[] {'R', 'e', 'q', 'u', 'e', 's', 't'};
+  private static final byte[] TEST_RESPONSE = new byte[] {'R', 'e', 's', 'p', 'o', 'n', 's', 'e'};
+  private static final byte[] TEST_ASSOCIATED_DATA = new byte[0];
 
+  private static class TestTransport implements EvidenceProvider, Transport {
     @Override
-    public Result<PilotRpcClient, Exception> getRpcClient() {
-      return Result.success(rpcClient);
+    public Result<AttestationBundle, String> getEvidence() {
```

Comment:
Why is the error type `String` here?

Reply:
Updated after this discussion: https://github.com/project-oak/oak/pull/3877#discussion_r1171120500